runtime/sema/account_contracts.go

```diff
@@ -0,0 +1,88 @@
+/*
```

Comment:
There was lots of duplication between `AuthAccount` and `PublicAccount` for the common functionality (get, names, borrow), so refactored this as part of this PR, so we don't add to the duplication